plugins/inputs/elasticsearch_query/aggregation_query.go

```diff
@@ -0,0 +1,203 @@
+package elasticsearch_query
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	elastic "gopkg.in/olivere/elastic.v5"
+)
+
+type aggKey struct {
+	measurement string
+	name        string
+	function    string
+	field       string
+}
+
+type aggregationQueryData struct {
+	aggKey
+	isParent    bool
+	aggregation elastic.Aggregation
+}
+
+func (e *ElasticsearchQuery) runAggregationQuery(ctx context.Context, aggregation esAggregation) (*elastic.SearchResult, error) {
+
+	now := time.Now().UTC()
+	from := now.Add(aggregation.QueryPeriod.Duration * -1)
```

Comment:
Still open...

Reply:
I noticed I cannot just replace as suggested because `Sub` requires a `Time` as `Add` requires a `Duration`

Reply:
Ok than at least make it
```suggestion
	from := now.Add(-aggregation.QueryPeriod.Duration)
```